org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/xsd/participants/diagnostics/XSDValidator.java

```diff
@@ -57,50 +69,59 @@ public static void doDiagnostics(DOMDocument document, XMLEntityResolver entityR
 			grammarPreparser.setFeature(Constants.XERCES_FEATURE_PREFIX + Constants.WARN_ON_DUPLICATE_ATTDEF_FEATURE,
 					true);
 
-			/*
-			 * if(configuration.getFeature(XSDValidationConfiguration.
-			 * HONOUR_ALL_SCHEMA_LOCATIONS)) { try {
-			 * grammarPreparser.setFeature(Constants.XERCES_FEATURE_PREFIX +
-			 * "honour-all-schemaLocations", true); //$NON-NLS-1$ } catch (Exception e) { //
-			 * catch the exception and ignore } }
-			 * 
-			 * if(configuration.getFeature(XSDValidationConfiguration.
-			 * FULL_SCHEMA_CONFORMANCE)) { try {
-			 * grammarPreparser.setFeature(Constants.XERCES_FEATURE_PREFIX +
-			 * Constants.SCHEMA_FULL_CHECKING, true); } catch (Exception e) { // ignore
-			 * since we don't want to set it or can't. }
-			 * 
-			 * }
-			 */
-
 			// Add LSP content handler to stop XML parsing if monitor is canceled.
 			// grammarPreparser.setContentHandler(new LSPContentHandler(monitor));
 
 			// Add LSP error reporter to fill LSP diagnostics from Xerces errors
-			grammarPreparser.setProperty("http://apache.org/xml/properties/internal/error-reporter",
-					new LSPErrorReporterForXSD(document, diagnostics));
+			grammarPreparser.setProperty("http://apache.org/xml/properties/internal/error-reporter", reporter);
 
 			if (entityResolver != null) {
 				grammarPreparser.setEntityResolver(entityResolver);
 			}
 
-			try {
-				String content = document.getText();
-				String uri = document.getDocumentURI();
-				InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
-				XMLInputSource is = new XMLInputSource(null, uri, uri, inputStream, null);
-				grammarPreparser.getLoader(XMLGrammarDescription.XML_SCHEMA);
-				grammarPreparser.preparseGrammar(XMLGrammarDescription.XML_SCHEMA, is);
-			} catch (Exception e) {
-				// parser will return null pointer exception if the document is structurally
-				// invalid
-				// TODO: log error message
-				// System.out.println(e);
-			}
+			String content = document.getText();
+			String uri = document.getDocumentURI();
+			InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+			XMLInputSource is = new XMLInputSource(null, uri, uri, inputStream, null);
+			grammarPreparser.getLoader(XMLGrammarDescription.XML_SCHEMA);
+			grammarPreparser.preparseGrammar(XMLGrammarDescription.XML_SCHEMA, is);
+		} catch (IOException | CancellationException exception) {
+			// ignore error
 		} catch (Exception e) {
-			// TODO: log error.
-			// System.out.println(e);
+			LOGGER.log(Level.SEVERE, "Unexpected XMLValidator error", e);
 		}
 	}
 
+	/**
+	 * Create the XML Schema loader to use to validate the XML Schema.
+	 * 
+	 * @param reporter the lsp reporter.
+	 * @return the XML Schema loader to use to validate the XML Schema.
+	 * @throws NoSuchFieldException
+	 * @throws IllegalAccessException
+	 */
+	private static XMLSchemaLoader createSchemaLoader(XMLErrorReporter reporter)

```

Comment:
if reflection fails once, it should not be reattempted, XSD validation should be disabled

Reply:
@fbricon updated